src/request.rs

```diff
@@ -345,16 +351,26 @@ fn parse_response<Output: DeserializeOwned>(
             }
         };
     }
-    // TODO: create issue where it is clear what the HTTP error is
-    // ParseError(Error("invalid type: null, expected struct MeilisearchError", line: 1, column: 4))
 
     warn!(
         "Expected response code {}, got {}",
         expected_status_code, status_code
     );
+
     match from_str::<MeilisearchError>(&body) {
         Ok(e) => Err(Error::from(e)),
-        Err(e) => Err(Error::ParseError(e)),
+        Err(e) => {
+            if status_code >= 400 {
+                return Err(Error::MeilisearchCommunication(
+                    MeilisearchCommunicationError {
+                        status_code,
+                        message: None,
+                        url,
+                    },
+                ));
+            }
+            Err(Error::ParseError(e))
+        }
```

Comment:
We've talked about this, but as a reminder, the central issue of the error handler is this:
https://github.com/meilisearch/integration-guides/issues/19
```
Following the previous issue about error handler naming, here are the 3 minimum errors the SDKs should raise:

- MeiliSearchApiError: MeiliSearch (HTTP) sends back a 4xx or a 5xx. There are most of the SDKs errors.
- MeiliSearchCommunicationError: problem with communication, for example, the MeiliSearch instance is unreachable.
- MeiliSearchError: for any other errors in the SDK (wrong parameter if the SDK checks the parameters...etc)
```
I know that for you it should be in the MeilisearchCommunicationError but this is not the case in most SDK. It might be worth opening an issue about this in integration-guide
WDYT? I can do it if you want